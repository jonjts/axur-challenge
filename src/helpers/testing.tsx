import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import ThemeProvider from '../providers/Theme';
import InspectionProvider from '../providers/Inspection'

export const putProviders = (component: JSX.Element) => (
    <ThemeProvider>
        <InspectionProvider>
            {component}
        </InspectionProvider>
    </ThemeProvider>
);

export const matchSnapshotWithProviders = (component: JSX.Element) => {
    const tree = renderer.create(putProviders(component)).toJSON();
    expect(tree).toMatchSnapshot();
    return tree;
};

export const renderWithProviders = (component: JSX.Element) => {
    return render(putProviders(component));
};
