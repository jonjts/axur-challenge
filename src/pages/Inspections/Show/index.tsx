import React, { useState, useEffect, useCallback } from 'react';
import { useHistory, useParams } from 'react-router-dom';

import { InspectionResult } from '../../../entities';
import { useInspection } from '../../../providers/Inspection';

import Spinner from '../../../components/Spinner';
import Card from '../../../components/Card';
import Button from '../../../components/Button';

import Results from './Results';

import {
  CardHeader,
  CardBody,
  SpinnerContainer,
  Content
} from './styles';

const Show: React.FC = () => {
  const { goBack } = useHistory();
  const { find, loading } = useInspection();
  const { id } = useParams() as { id?: string };

  const [inspectionId, setInspectionId] = useState<string>(id || '');
  const [inspectionResult, setInspectionResult] = useState({} as InspectionResult);

  const findInspection = useCallback(async () => {
    setInspectionResult({} as InspectionResult);
    const result = await find(inspectionId.trim());
    setInspectionResult(result);
  }, [find, inspectionId]);

  useEffect(() => {
    if (id) findInspection();
  }, [id]);

  return (
    <>
      <h1>Ver Inspeção</h1>
      <Card
        showBackButton
        backText='Voltar'
        onBackClicked={goBack}
        disableButtons={loading}
      >
        <CardHeader>
          {
            !id ? (
              <>
                <input
                  value={inspectionId}
                  onChange={({ target: { value } }) => setInspectionId(value)}
                  placeholder='Informe o id da inspeção'
                />
                <Button onClick={findInspection}>Pesquisar</Button>
              </>
            ) : (
              <h2>
                ID: <i>{id}</i>
              </h2>
            )
          }
        </CardHeader>
        <CardBody className='row'>
          {
            loading ? (
              <SpinnerContainer className='row'>
                <Spinner />
              </SpinnerContainer>
            ) : (
              <Content className='grid'>
                {inspectionResult?.id && <Results result={inspectionResult} />}
              </Content>
            )
          }
        </CardBody>
      </Card>
    </>
  );
};

export default Show;