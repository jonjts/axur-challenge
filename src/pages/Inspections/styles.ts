import styled from 'styled-components';

export const Container = styled.div`
`;

export const Content = styled.div`
    @media (min-width: 992px) { 
        grid-template-columns: 32% 32% 32% !important;
    }
`;

export const CardContent = styled.div`
    display: flex;
    flex: 1;
    flex-direction: column;

    small {
        font-size: 11px;
        align-self: flex-end;
        position: absolute;
        bottom: 10px;
        right: 10px;
    }

    h3 {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
`;