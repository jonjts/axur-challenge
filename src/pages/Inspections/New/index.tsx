import React, { useState, useCallback, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

import { useInspection } from '../../../providers/Inspection';

import Card from '../../../components/Card';

import { FormGroup } from './styles';

const New: React.FC = () => {
  const { goBack } = useHistory();
  const { insert, loading } = useInspection();

  const [keyword, setKeyword] = useState('');
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setHasError(false);
  }, [keyword]);

  const handleSave = useCallback(async () => {
    try {
      await insert(keyword);
      goBack();
    } catch (error) {
      setHasError(true);
    }
  }, [insert, keyword, goBack]);


  return (
    <>
      <h1>Nova Inspeção</h1>
      <Card
        showBackButton
        showSubmitButton
        backText='Voltar'
        submitText='Salvar'
        onBackClicked={goBack}
        onSubmitClicked={handleSave}
        disableButtons={loading}
      >
        <FormGroup hasError={hasError}>
          <label>Keyword</label>
          <input
            placeholder='Informe a keyword'
            value={keyword}
            onChange={({ target: { value } }) => setKeyword(value)}
          />
        </FormGroup>
      </Card>
    </>
  );
};

export default New;