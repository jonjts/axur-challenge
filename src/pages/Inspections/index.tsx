import React from 'react';
import { useHistory } from 'react-router-dom';

import Button from '../../components/Button';

import { Container } from './styles';

const Inspections: React.FC = () => {
  const { push } = useHistory();

  return (
    <Container>
      <h1>Inspeções</h1>
      <Button onClick={() => push('inspections/new')}>
        Nova Inspeção
      </Button>
    </Container>
  );
};

export default Inspections;