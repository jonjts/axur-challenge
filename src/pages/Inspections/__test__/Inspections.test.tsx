import { matchSnapshotWithProviders, renderWithProviders } from '../../../helpers/testing';

import Inspections from '../index';

describe('Inspections', () => {
  test('match with snapshot', () => {
    matchSnapshotWithProviders(<Inspections />);
  });

  test('renders correctly', () => {
    renderWithProviders(<Inspections />);
  });
});
