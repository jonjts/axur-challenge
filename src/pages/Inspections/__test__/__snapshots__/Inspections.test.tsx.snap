// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inspections match with snapshot 1`] = `
<div
  className="sc-jSgupP eplsmV"
>
  <h2>
    Inspeções Recentes
  </h2>
  <div
    className="sc-gKsewC gXHAJb grid"
  />
</div>
`;
